src/libs/events.js

```diff
@@ -63,4 +65,48 @@ export const PageViewReporter = () => {
   return null
 }
 
+export const captureAppcuesEvent = (eventName, event) => {
+  // Only record "public-facing events" (and related properties) as documented by Appcues: https://docs.appcues.com/article/301-client-side-events-reference
+  const publicEvents = [
+    'flow_started',
+    'flow_completed',
+    'flow_skipped',
+    'flow_aborted',
+    'step_started',
+    'step_completed',
+    'step_skipped',
+    'step_aborted',
+    'step_interacted',
+    'form_submitted',
+    'form_field_submitted'
+  ]
+  if (_.includes(eventName, publicEvents)) {
+    const eventProps = {
+      'appcues.flowId': event.flowId,
+      'appcues.flowName': event.flowName,
+      'appcues.flowType': event.flowType,
+      'appcues.flowVersion': event.flowVersion,
+      'appcues.id': event.id,
+      'appcues.interaction.category': event.interaction?.category,
+      'appcues.interaction.destination': event.interaction?.destination,
+      'appcues.interaction.element': event.interaction?.element,
+      'appcues.interaction.fields': event.interaction?.fields && JSON.stringify(event.interaction.fields),
```

Comment:
Actually, prodded a bit deeper, turns out that `JSON.stringify(undefined) === undefined`, so you can just
```suggestion
      'appcues.interaction.fields': JSON.stringify(event.interaction?.fields),
```